floodsub_test.go

```diff
@@ -943,3 +947,120 @@ func TestWithSigning(t *testing.T) {
 		t.Fatalf("unexpected data: %s", string(msg.Data))
 	}
 }
+
+func TestImproperlySignedMessageNotRelayed(t *testing.T) {
```

Comment:
We create two hosts so we have the following topology:
```
adversary <-> honest
```
and they are subscribed to the same topic named `"foobar"`.

In that case, I think this test is only about rejecting a message by `honest` peer. It's not about rejecting relayed message so it should be named `TestImproperlySignedMessageRejected`.

I also think we need another test if we want to have our network security requirements/assumptions covered. We need the following network topology:
```
adversary <-> relay_peer <-> destination_peer
```
- `adversary` should be subscribed to topic `"foobar"`
- `destination_peer` should be subscribed to topic `"foobar"`
- `relay_peer` should NOT be subscribed to any topic
- `adversary` sends a correct message to `destination_peer` and we need to make sure it's received
- `adversary` sends an incorrect message to `destination_peer` and we need to make sure it's rejected by `relay_peer`.